### PR TITLE
add eks and ecs exec types for golang

### PR DESCRIPTION
**Pull request by jhsinger-klotho**
• Does any part of it require special attention?
• Does it relate to or fix any issue? closes https://github.com/klothoplatform/klotho-history/issues/469

Images are only 25-35 MB for EKS and ECS with golang. 

Adding new dockerfile and doing some refactors to the expose plugin to not be lambda specific/moving that into runtime

exporting aws service names from the provider so they can be used in the runtime

### Standard checks

- **Unit tests**: Any special considerations? all are still passing, shouldnt need anything new since the functionality is all infra based
- **Docs**: Do we need to update any docs, internal or public? not needed
- **Backwards compatibility**: Will this break existing apps? If so, what would be the extra work required to keep them working? yes

### CODE REVIEW

**Review by gordon-klotho — APPROVED** [review]
Nothing blocking, but some improvements in the comments

File: pkg/lang/golang/aws_runtime/aws.go [hunk 1]
```diff
@@ -58,6 +66,83 @@ func (r *AwsRuntime) AddExecRuntimeFiles(unit *core.ExecutionUnit, result *core.
...
+			}
+			lambda.StartWithContext(context.Background(), handler)
+			//End - Added by Klotho`, routerName)
+
+			newNodeContent = newNodeContent + dispatcherCode
+			newFileContent = strings.ReplaceAll(
+				newFileContent,
+				oldNodeContent,
+				newNodeContent,
+			)
```
[the remark this answers is not in the document]
Comment (jhsinger-klotho) [reply pkg/lang/golang/aws_runtime/aws.go:101]: this was copy paste from the initial code, just moved it into runtime, so let me refactor a bit to clean it up

File: pkg/lang/golang/aws_runtime/Exec_Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM golang:alpine as builder
+RUN apk update && apk add ca-certificates && rm -rf /var/cache/apk/*
+
+WORKDIR /usr/src/app
+ENV GOOS=linux GOARCH=amd64 CGO_ENABLED=0
+COPY . .
```
[the remark this answers is not in the document]
Comment (jhsinger-klotho) [reply pkg/lang/golang/aws_runtime/Exec_Dockerfile:6]: so the previous pr you had put in for the dockerfile didnt fully work until i made some of the changes which are now reflected in both spots. I can try to make the change your suggesting but if i remember correctly, it wasnt building the image as expected

[the remark this answers is not in the document]
Comment (gordon-klotho) [reply pkg/lang/golang/aws_runtime/Exec_Dockerfile:6]: Yeah, the layer caching is mostly for build times and can be a huge help for CI/CD cases but doesn't affect the runtime image in this case.

**Review by gordon-klotho — APPROVED** [review]

**Pull request merged**